### PR TITLE
Better handle Socket transport accept errors

### PR DESCRIPTION
**Pull request by halter73**
Also don't log server-aborted connections as errors.

### CODE REVIEW

**Comment by halter73** [comment]
This should fix the flaky ConnectionFilterDoesNotLeakBlock test.

File: src/Kestrel.Transport.Sockets/SocketTransport.cs [hunk 1]
```diff
@@ -89,14 +94,22 @@ public Task BindAsync()
...
                 Debug.Assert(_listenTask != null);
                 await _listenTask.ConfigureAwait(false);
+
+                _unbinding = false;
+                _listenSocket = null;
                 _listenTask = null;
+
+                if (_listenException != null)
+                {
+                    var exInfo = ExceptionDispatchInfo.Capture(_listenException);
```
Comment (anurse) [inline src/Kestrel.Transport.Sockets/SocketTransport.cs]: What's the point of capturing and then immediately rethrowing? Aren't you supposed to `.Capture` at the point where the exception occurs and then `.Throw`?

I may be misinterpreting how EDI works though.

Comment (pakrym) [reply src/Kestrel.Transport.Sockets/SocketTransport.cs]: You don't have to capture at the point of exception. See https://github.com/dotnet/corefx/pull/17570

File: src/Kestrel.Transport.Sockets/SocketTransport.cs [hunk 2]
```diff
@@ -112,23 +125,39 @@ public Task StopAsync()
...
+                        var acceptSocket = await _listenSocket.AcceptAsync();
+                        acceptSocket.NoDelay = _endPointInformation.NoDelay;
+
+                        var connection = new SocketConnection(acceptSocket, _pipeFactory, _trace);
+                        _ = connection.StartAsync(_handler);
+                    }
+                    catch (SocketException ex) when (ex.SocketErrorCode == SocketError.ConnectionReset)
+                    {
+                        // REVIEW: Should there be a seperate log message for a connection reset this early?
+                        _trace.ConnectionReset(connectionId: "(null)");
```
Comment (pakrym) [inline src/Kestrel.Transport.Sockets/SocketTransport.cs]: Do you want to log connection reset when unbinding?

File: src/Kestrel.Transport.Sockets/SocketTransport.cs (hunk 1 above)
Comment (anurse) [reply src/Kestrel.Transport.Sockets/SocketTransport.cs]: I withdraw my concerns then :). Thanks for clarifying.

File: src/Kestrel.Transport.Sockets/SocketTransport.cs (hunk 2 above)
Comment (halter73) [reply src/Kestrel.Transport.Sockets/SocketTransport.cs]: Yes. AFAICT unbinding doesn't cause a SocketError.ConnectionReset. If it happens to occur during shutdown, I don't see the harm in this info level log message.

**Pull request merged**